### DISCUSSION TITLE
body()

### DISCUSSION THREAD

**Discussion by samaanjafari** [fastapi/fastapi#11023] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Body, FastAPI

@app.post("/createposts")
def create_post(payload : dict =  Body(...)):     
    print(payload)
    return {"new_post": f"title: {payload['title']} content: {payload['content']}"}
```

### Description

can someone tell me about differences between 3dots in parentheses in body(...) and body() documentation is saying that 3dots is for validation if we aren't using pydantic models and body(...) will handle the validation itself if for example in our request body some parameters were missing body will take of it but i tested both of them body(...) and body() but neither of them do anything at couldn't understand their diffrences.

### Operating System

Windows

### FastAPI Version

0.105.0

### Pydantic Version

2.5.3

### Python Version

3.11.0

**Comment by Kludex** [comment]
Can you show me an example on a call that is not doing what is supposed to do?

**Reply by samaanjafari** [reply]
actually i don't know what i supposed to see exactly as i have been understood from documents ellipsis should role as our pydantic models if we haven't set them yet but it really doesn't change anything if ellipsis will be there our not if we pass an unvalid type or missed some parameter in our request body it will pops up an error though. so  i was just asking is ellipsis there are useless?

**Comment by bogdan-coman-uv** [comment]
If your body is a plain dict, there will be no validation. You need to use a pydantic model or a dataclass for the validation to work. In pydantic models or dataclasses you can specify which fields are required, the expected type for each field, allowed values and so on

**Reply by samaanjafari** [reply]
yess i knew that i have to use pydantic models. i just wanted to know what is purpose of ellipsis in body parameter and what it actually do.
